pkg/apis/agones/v1/gameserver_test.go

```diff
@@ -543,18 +549,29 @@ func TestGameServerPod(t *testing.T) {
 	assert.True(t, metav1.IsControlledBy(pod, fixture))
 	assert.Equal(t, fixture.Spec.Ports[0].HostPort, pod.Spec.Containers[0].Ports[0].HostPort)
 	assert.Equal(t, fixture.Spec.Ports[0].ContainerPort, pod.Spec.Containers[0].Ports[0].ContainerPort)
+	assert.Equal(t, fixture.Spec.Ports[0].ContainerName, pod.Spec.Containers[0].Name)
 	assert.Equal(t, corev1.Protocol("UDP"), pod.Spec.Containers[0].Ports[0].Protocol)
 	assert.True(t, metav1.IsControlledBy(pod, fixture))
 
 	sidecar := corev1.Container{Name: "sidecar", Image: "container/sidecar"}
 	fixture.Spec.Template.Spec.ServiceAccountName = "other-agones-sdk"
+	fixture.Spec.Ports = append(fixture.Spec.Ports, GameServerPort{
```

Comment:
Do we want to split this into two tests? Once with the default port setup, and one with this new functionality - just to get coverage on both scenarios?